Assets/__Scripts/UI/DevConsole.cs

```diff
@@ -0,0 +1,198 @@
+        SceneManager.sceneLoaded -= SceneLoaded;
+    }
+    
+    private void SceneLoaded(Scene arg0, LoadSceneMode arg1)
+    {
+        var yPos = arg0.name.Contains("Mapper") ? 30 : 10;
+        GetComponent<RectTransform>().anchoredPosition = new Vector3(10, yPos, 0);
+    }
+
+    public void Update()
+    {
+        while (_backlog.TryDequeue(out var logline)) ShowLogline(logline);
+    }
+
+    private void FixedUpdate()
+    {
+        _writer?.Flush();
+    }
+
+    private readonly Dictionary<LogType, string> _logColors = new Dictionary<LogType, string>()
+    {
+        {LogType.Log, "#FFFFFF"},
+        {LogType.Assert, "#32AD10"},
+        {LogType.Error, "#F02B2B"},
+        {LogType.Exception, "#AF3DFF"},
+        {LogType.Warning, "#EBCF34"}
+    };
+
+    private void LogCallback(string condition, string stackTrace, LogType type)
+    {
+        ShowLogline(new Logline(type, condition, stackTrace));
+    }
+
+    private void ShowLogline(Logline logline)
+    {
+        if (logline.Type == LogType.Error || logline.Type == LogType.Exception)
+        {
+            scrollRect.gameObject.SetActive(true);
+        }
+        
+        _lines.Add(logline.Message);
+        _writer.WriteLine("[" + logline.Type + "] " + logline.Message);
+        if (!string.IsNullOrWhiteSpace(logline.StackTrace))
+        {
+            _writer.WriteLine(logline.StackTrace);
+        }
+
+        LogLineUI newElement;
+        if (_uiElements.Count >= MaxLines)
+        {
+            newElement = _uiElements[0];
+            _uiElements.RemoveAt(0);
+            newElement.transform.SetAsLastSibling();
+        }
+        else
+        {
+            newElement = Instantiate(logRow, rowParent);
+        }
+        _uiElements.Add(newElement);
+
+        newElement.gameObject.SetActive(true);
+        newElement.SetupReport(logline, _lines);
+        newElement.TextMesh.text = $"<color={_logColors[logline.Type]}>" + logline.Message + "</color>\n";
+        StopCoroutine(nameof(ScrollToBottom));
+        StartCoroutine(nameof(ScrollToBottom));
+    }
+
+    private IEnumerator ScrollToBottom()
+    {
+        yield return new WaitForEndOfFrame();
+        scrollRect.verticalNormalizedPosition = 0f;
+    }
+
+    public void Clear()
+    {
+        _lines.Clear();
+        foreach (var textMeshProUGUI in _uiElements)
+        {
+            textMeshProUGUI.gameObject.SetActive(false);            
+        }
+        StopCoroutine(nameof(ScrollToBottom));
+        StartCoroutine(nameof(ScrollToBottom));
+    }
+
+    private string GetLogFolder()
+    {
+#if UNITY_STANDALONE_WIN
+        return Path.Combine(Environment.GetEnvironmentVariable("AppData"), "..", "LocalLow", Application.companyName, Application.productName);
+#elif UNITY_STANDALONE_OSX
+        return "~/Library/Logs/Unity";
+#elif UNITY_STANDALONE_LINUX
+        return Path.Combine("~/.config/unity3d", Application.companyName, Application.productName);
+#endif
+    }
```

Comment:
Couldn't this all be simplified to a simple call to [`Application.persistentDataPath`](https://docs.unity3d.com/ScriptReference/Application-persistentDataPath.html)?

Reply:
This may be a derp moment